ocrd_network/ocrd_network/processing_server.py

```diff
@@ -0,0 +1,289 @@
+        self.rmq_publisher = RMQPublisher(
+            host=self.rmq_host,
+            port=self.rmq_port,
+            vhost=self.rmq_vhost
+        )
+        self.log.debug(f'RMQPublisher authenticates with username: '
+                       f'{self.rmq_username}, password: {self.rmq_password}')
+        self.rmq_publisher.authenticate_and_connect(
+            username=self.rmq_username,
+            password=self.rmq_password
+        )
+        if enable_acks:
+            self.rmq_publisher.enable_delivery_confirmations()
+            self.log.info('Delivery confirmations are enabled')
+        self.log.info('Successfully connected RMQPublisher.')
+
+    def create_message_queues(self) -> None:
+        """Create the message queues based on the occurrence of `processor.name` in the config file
+        """
+        for host in self.config.hosts:
+            for processor in host.processors:
+                # The existence/validity of the processor.name is not tested.
+                # Even if an ocr-d processor does not exist, the queue is created
+                self.log.info(f'Creating a message queue with id: {processor.name}')
+                self.rmq_publisher.create_queue(queue_name=processor.name)
+
+    @property
+    def processor_list(self):
+        if self._processor_list:
+            return self._processor_list
+        res = set([])
+        for host in self.config.hosts:
+            for processor in host.processors:
+                res.add(processor.name)
+        self._processor_list = list(res)
+        return self._processor_list
+
+    async def push_processor_job(self, processor_name: str, data: JobInput) -> JobOutput:
+        """ Queue a processor job
+        """
+        if processor_name not in self.processor_list:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail='Processor not available'
+            )
+
+        # validate additional parameters
+        if data.parameters:
+            ocrd_tool = get_ocrd_tool_json(processor_name)
+            if not ocrd_tool:
+                raise HTTPException(
+                    status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                    detail=(f'Processor \'{processor_name}\' not available. It\'s ocrd_tool is '
+                            'empty or missing')
+                )
+            validator = ParameterValidator(ocrd_tool)
+            report = validator.validate(data.parameters)
+            if not report.is_valid:
+                raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=report.errors)
+
+        # determine path to mets if workspace_id is provided
+        if bool(data.path) == bool(data.workspace_id):
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail='Either \'path\' or \'workspace_id\' must be set'
```

Comment:
The error message is misleading in case **both** `path` and `workspace_id` are set.

Reply:
By definition of the `OcrdProcessingMessage`, either of the two is set, never both.

Reply:
Ah, yes!